addons/account/views/account_payment_view.xml

```diff
@@ -281,10 +281,11 @@
                                             'readonly': [('state', '!=', 'draft')],
                                         }"/>
 
-                                <field name="partner_bank_id" context="{'default_partner_id': partner_id}" string="Company Bank Account" readonly="1"
+                                <field name="partner_bank_id" context="{'default_partner_id': partner_id}" string="Company Bank Account"
                                         attrs="{
                                             'invisible': ['|', '|', ('show_partner_bank_account', '=', False), ('is_internal_transfer', '=', True), ('payment_type', '=', 'outbound')],
                                             'required': [('require_partner_bank_account', '=', True), ('is_internal_transfer', '=', False)],
+                                            'readonly': [('payment_type', '!=', 'outbound')],
```

Comment:
Needs a condition on the state too

Reply:
I'm not sure as I think the intended behavior in this case is that the bank account will automatically be computed based on the journal chosen.
If I add a condition on the state like:
```xml
'readonly': [('state', '!=', 'draft'), ('payment_type', '!=', 'outbound')]
```
Then it will be possible to edit the bank account used but there is only a single bank account per journal so it will not be possible to change the bank account as there only is a single choice (see https://watch.screencastify.com/v/4bpuR4T4yfDlP9vq1w7c)

Do you agree ?

Reply:
Making it always readonly will also impact the previous `partner_bank_id` fields and put them in readonly too (making it impossible to edit the 'Customer Bank Account' and 'Vendor Bank Account')

Reply:
My understanding was that the attrs are evaluated on each individual field, but setting `readonly="1"` sets it on all the different instances of the field

Reply:
The field is indeed editable in the view when using readonly in attrs but changing the account will not do anything as the field will be detected as readonly in this function https://github.com/odoo/odoo/blob/f5ba34e45b66e0fccc2559bf434e255d38a0faa0/addons/web/static/src/legacy/js/views/basic/basic_model.js#L3911
So the modification will not be taken into account when saving the record

Reply:
Ok, so actually the `readonly` attr is shared, unlike the `invisible` attr.
So your change is actually setting `[('payment_type', '!=', 'outbound')]` on all the other instances of `partner_bank_id`, effectively overriding `[('state', '!=', 'draft')]` previously set elsewhere.

This said, I don't know why the field was set to be readonly; you can still have multiple bank accounts on the company
https://github.com/odoo/odoo/blob/4624d1ebdedf84ed3031f7af1be827de54f72566/addons/account/models/account_payment.py#L391-L392
@oco-odoo you wrote this https://github.com/odoo/odoo/commit/decf107dcc7d3b04978958563f2c30ccd0b1dba9 can you maybe give us more insight?